File: code/java/src/main/java/fr/zomzog/prez/javaVsKotlin/java/lambda/Lambda.java
package fr.zomzog.prez.javaVsKotlin.java.lambda;

import java.util.function.BiFunction;

public class Lambda {

    public String doStuff(String name, BiFunction<String, Integer, String> fun) {
        return fun.apply(name, 42);
    }
}

File: code/kotlin/src/main/kotlin/fr/zomzog/prez/javaVsKotlin/kotlin/lambda/Lambda.kt
package fr.zomzog.prez.javaVsKotlin.kotlin.lambda

//fun doStuff(list: List<Int>, f: (Int) -> Int) = list.map { it*2 }.map(f)
//fun doStuff(list: List<Int>, f: (Int, Int) -> Int) = list.map { it*2 }.map{f(it,it)}


fun doStuff(name: String, f: (String, Int) -> String): String {
    return f(name, 42)
}